### PR TITLE
ncm: add a few IPC wrappers for content storage and content meta data database

### PR DESCRIPTION
**Pull request by zhuowei**

### CODE REVIEW

File: nx/include/switch/services/ncm.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/**
+ * @file ncm.h
+ * @brief Content Manager (ncm) service IPC wrapper.
+ * @copyright libnx Authors
+ */
+#pragma once
+#include "../types.h"
+#include "../services/fs.h"
+#include "../services/sm.h"
+typedef enum NCMContentType {
```
Comment (yellows8) [inline nx/include/switch/services/ncm.h]: add newline

File: nx/include/switch/services/ncm.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+typedef struct {
+    u64 titleID;    // 0x0
+    u32 version;    // 0x8
+    u8  type;       // 0xc
+    u8  flags;      // 0xd
+    u8  padding[2]; // 0xe
+} NCMMetaRecord;
+
+typedef struct {
+    char c[0x10];
```
Comment (yellows8) [inline nx/include/switch/services/ncm.h]: u8

File: nx/source/services/ncm.c [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+#include "services/sm.h"
+#include "services/ncm.h"
+
+static Service g_ncmSrv;
+
+static u64 g_ncmRefCnt;
+
+Result ncmInitialize() {
+    atomicIncrement64(&g_ncmRefCnt);
+    Result rc = smGetService(&g_ncmSrv, "ncm");
```
Comment (yellows8) [inline nx/source/services/ncm.c]: Missing {already-initialized} code which other service code has.

File: nx/source/services/ncm.c [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+        if (R_SUCCEEDED(rc)) {
+            *out = resp->size;
+        }
+    }
+    
+    return rc;
+}
+
+Result ncmContentStorageReadContentIdFile(NCMContentStorage* cs, const NCMNCAID* ncaId, u64 offset, char* outBuf, size_t bufSize) {
```
Comment (yellows8) [inline nx/source/services/ncm.c]: void* outBuf

**Comment by fincs** [comment]
Any updates regarding the above changes?

File: nx/include/switch/services/ncm.h [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+} NCMNCAID;
```
Comment (yellows8) [inline nx/include/switch/services/ncm.h]: NcmNcaId

**Comment by yellows8** [comment]
Change all "NCM" names to "Ncm".

**Comment by fincs** [comment]
Superseded by #129.

**Pull request closed without merging**